lib/spack/spack/installer.py

```diff
@@ -2424,6 +2425,12 @@ def _install(self) -> None:
         install_status = InstallStatus(len(self.build_pq))
         active_tasks: List[Task] = []
 
+        # Determine which type of jobserver to set up and then enable it
+        packages = [task.pkg for _, task in self.build_pq]
+        jobserver_class = spack.jobserver.Jobserver.determine_type(packages)
```

Comment:
Does the jobserver type only get set once for the parent install processes of the DAG? So if I have a build dependency on python and need to install it, will the jobserver be disabled for the entire build?